packages/connect/src/protocol/normalize.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { PartialMessage, MessageType } from "@bufbuild/protobuf";
+import { Message } from "@bufbuild/protobuf";
+import { ConnectError } from "../connect-error.js";
+import { Code } from "../code.js";
+
+/**
+ *  Takes a partial protobuf messages of the
+ *  specified message type as input, and returns full instances.
+ */
+export function normalize<T extends Message<T>>(
+  type: MessageType<T>,
+  message: T | PartialMessage<T>,
+) {
+  if (message instanceof Message) {
+    return message;
+  }
+  try {
+    return new type(message);
+  } catch (e) {
+    throw new ConnectError(
+      `failed to normalize message ${type.typeName}`,
+      Code.Internal,
+      undefined,
+      undefined,
+      e,
+    );
+  }
```

Comment:
It's certainly not the most important thing - correctness is certainly more important - but 3/4 of the bundle size increase of this PR come from re-throwing the error here. Do you have a specific code path in mind where this could occur? We typically wouldn't recommend to wrap constructor calls with a try-catch.

Reply:
Ah, we had two version before. Yes, let's remove the re-throw.